BuildMacOSUniversalBinary.py

```diff
@@ -0,0 +1,234 @@
+# 2) Single architecture builds should utilize the normal generated cmake
+#    project files rather than this wrapper script
+
+architectures = ["x86_64", "arm64"]
+
+# Minimum macOS version for each architecture slice
+mac_os_deployment_target={
+    "arm64": "10.14.0",
+    "x86_64": "10.12.0"
+}
+
+def parse_args():
+    global dst_app, build_target, pkg_config_path, qt5_path, codesign_identity
+
+    parser = argparse.ArgumentParser(formatter_class=
+                                     argparse.ArgumentDefaultsHelpFormatter)
+
+    parser.add_argument('--target',
+        help='Build target in generated project files',
+        default=build_target)
+
+    parser.add_argument('--dst_app',
+        help='Directory where universal binary will be stored',
+        default=dst_app)
+
+    parser.add_argument('--codesign',
+        help='Code signing identity to use to sign the applications',
+        default=codesign_identity)
+
+    for arch in architectures:
+        parser.add_argument('--{}_pkg_config'.format(arch),
+             help="Folder containing .pc files for {} libraries".format(arch),
+             default=pkg_config_path[arch])
+
+        parser.add_argument('--{}_qt5_path'.format(arch),
+             help="Install path for {} qt5 libraries".format(arch),
+             default=qt5_path[arch])
+
+    args = vars(parser.parse_args());
+    dst_app = args["dst_app"]
+    build_target = args["target"]
+    codesign_identity = args["codesign"]
+
+    for arch in architectures:
+        pkg_config_path[arch] = args["{}_pkg_config".format(arch)]
+        qt5_path[arch] = args["{}_qt5_path".format(arch)]
+
+
+def lipo(path0,path1,dst):
+    if subprocess.call(['lipo', '-create', '-output', dst, path0, path1]) != 0:
+        print("WARNING: {} and {} are different but can not be lipo'd, keeping {}".format(path0, path1, path0));
+        shutil.copy(path0, dst)
+
+# Merges two build trees together for different architectures into a single
+# universal binary.
+# The rules for merging are:
+# 1) Files that exist in either src tree are copied into the dst tree
+# 2) Files that exist in both trees and are identical are copied over
+#    unmodified
+# 3) Files that exist in both trees and are non-identical are lipo'd
+# 4) Symlinks are created in the destination tree to mirror the hierarchy in
+#    the source trees
+
+def recursiveMergeBinaries(src0,src1,dst):
+    #loop over all files in src0
+    for newpath0 in glob.glob(src0+"/*"):
+        filename = os.path.basename(newpath0);
+        newpath1 = os.path.join(src1,filename);
+        new_dst_path = os.path.join(dst,filename);
+        if not os.path.islink(newpath0):
+            if os.path.exists(newpath1):
+                if os.path.isdir(newpath1):
+                    os.mkdir(new_dst_path);
+                    #recurse into directories
+                    recursiveMergeBinaries(newpath0,newpath1,new_dst_path)
+                else:
```

Comment:
`else` block can be unindented by using `continue` after the `recursiveMergeBinaries` call.

Reply:
Done